Rename Tensor::names() to Tensor::opt_names()

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25123 Implement name inference for torch.bmm
* #24474 Implement name inference for torch.dot
* #24471 Implement name inference for addmv, addmv_, mv
* #24469 Implement name inference for expand
* #24306 Implement name inference for mm, addmm, addmm_
* #24921 Fix binary op name inference between unnamed and named tensors.
* #24914 Add helper function Tensor::names()
* **#24907 Rename Tensor::names() to Tensor::opt_names()**
* #24885 Fix `FIXME_default_names` by storing static list of 64 none names
* #24942 Add thread-local-state NamesMode and NoNamesGuard
* #24941 Implement name inference for t(), transpose(...)

This better reflects the semantics because Tensor::opt_names() returns
an `optional<DimnameList>`, not just a DimnameList.

Also rename `impl::get_names` to `impl::get_opt_names` (that is the
`TensorImpl*` variant of `Tensor::opt_names()`.

Test Plan
- run existing tests [namedtensor ci]

gh-metadata: pytorch pytorch 24907 gh/zou3519/110/head

Differential Revision: [D16919768](https://our.internmc.facebook.com/intern/diff/D16919768)

zou3519 committed:
Rename Tensor::names() to Tensor::opt_names()

This better reflects the semantics because Tensor::opt_names() returns
an optional<DimnameList>.

Test Plan
- run existing tests [namedtensor ci]

zou3519 committed:
Update on "Rename Tensor::names() to Tensor::opt_names()"

Rename Tensor::names() to Tensor::opt_names()

This better reflects the semantics because Tensor::opt_names() returns
an `optional<DimnameList>`, not just a DimnameList.

Also rename `impl::get_names` to `impl::get_opt_names` (that is the
`TensorImpl*` variant of `Tensor::opt_names()`.

Test Plan
- run existing tests [namedtensor ci]

gh-metadata: pytorch pytorch 24907 gh/zou3519/110/head

zou3519 committed:
Update on "Rename Tensor::names() to Tensor::opt_names()"

Rename Tensor::names() to Tensor::opt_names()

This better reflects the semantics because Tensor::opt_names() returns
an `optional<DimnameList>`, not just a DimnameList.

Also rename `impl::get_names` to `impl::get_opt_names` (that is the
`TensorImpl*` variant of `Tensor::opt_names()`.

Test Plan
- run existing tests [namedtensor ci]

gh-metadata: pytorch pytorch 24907 gh/zou3519/110/head

zou3519 committed:
Update on "Rename Tensor::names() to Tensor::opt_names()"

Rename Tensor::names() to Tensor::opt_names()

This better reflects the semantics because Tensor::opt_names() returns
an `optional<DimnameList>`, not just a DimnameList.

Also rename `impl::get_names` to `impl::get_opt_names` (that is the
`TensorImpl*` variant of `Tensor::opt_names()`.

Test Plan
- run existing tests [namedtensor ci]

gh-metadata: pytorch pytorch 24907 gh/zou3519/110/head

zou3519 committed:
Update on "Rename Tensor::names() to Tensor::opt_names()"

Rename Tensor::names() to Tensor::opt_names()

This better reflects the semantics because Tensor::opt_names() returns
an `optional<DimnameList>`, not just a DimnameList.

Also rename `impl::get_names` to `impl::get_opt_names` (that is the
`TensorImpl*` variant of `Tensor::opt_names()`.

Test Plan
- run existing tests [namedtensor ci]

gh-metadata: pytorch pytorch 24907 gh/zou3519/110/head

zou3519 committed:
Update on "Rename Tensor::names() to Tensor::opt_names()"

Rename Tensor::names() to Tensor::opt_names()

This better reflects the semantics because Tensor::opt_names() returns
an `optional<DimnameList>`, not just a DimnameList.

Also rename `impl::get_names` to `impl::get_opt_names` (that is the
`TensorImpl*` variant of `Tensor::opt_names()`.

Test Plan
- run existing tests [namedtensor ci]

gh-metadata: pytorch pytorch 24907 gh/zou3519/110/head

nairbv commented on aten/src/ATen/templates/Tensor.h:
```diff
@@ -173,8 +173,8 @@ class CAFFE2_API Tensor {
-  optional<DimnameList> names() const {
-    return impl::get_names(unsafeGetTensorImpl());
+  optional<DimnameList> opt_names() const {
+    return impl::get_opt_names(unsafeGetTensorImpl());
```
Could it make sense to do something like returning a default value instead, e.g. a DimnameList containing all wildcards if empty?  It seems for device() and scalar_type() we raise an error or return UndefinedType respectively when not set.

zou3519 replied:
Yes, exactly, see https://github.com/pytorch/pytorch/pull/24914 . I've changed it so that:

1) `Tensor::opt_names` gives `Optional<DimnameList>` (this PR)
2) `Tensor::names` gives a default DimnameList containing all wildcards if empty. (https://github.com/pytorch/pytorch/pull/24914)

zou3519 replied:
Returning the default value is a great idea because it makes writing name inference rules a lot easier.

zou3519 committed:
Update on "Rename Tensor::names() to Tensor::opt_names()"

Rename Tensor::names() to Tensor::opt_names()

This better reflects the semantics because Tensor::opt_names() returns
an `optional<DimnameList>`, not just a DimnameList.

Also rename `impl::get_names` to `impl::get_opt_names` (that is the
`TensorImpl*` variant of `Tensor::opt_names()`.

Test Plan
- run existing tests [namedtensor ci]

gh-metadata: pytorch pytorch 24907 gh/zou3519/110/head
